Technical-Steering-Committee/guidelines/breaking-changes.md

```diff
@@ -0,0 +1,136 @@
+    - Adding or removing errors
+    - Altering expected timing of an event
+
+**Examples of non-breaking changes:**
+
+1. CLI-specific non-breaking changes:
+
+    - Adding a command or a command group
+    - Adding optional argument
+    - Adding a response field to responses with two or more fields
+
+2. REST API-specific non-breaking changes:
+
+    - Adding a resource or a method
+    - Adding a response field
+    - Adding optional query parameters
+
+3. Programmatic API-specific breaking changes:
+
+    - Adding a class or a method
+    - Adding a optional parameter
+    - Adding a field to a class
+
+## Deprecations
+
+Zowe uses three Deprecation levels. For all deprecated APIs, the API documentation must state the deprecation status.
+
+1. Documentation-Only Deprecation
+
+    - A deprecation notice appears in the documentation in the API reference documentation and places where the functionality is described for the active and under development releases.
+    - In the case of programmatic Java or Node.js APIs, deprecation mechanisms provided in the programming language are used (e.g. `@Deprecated` annotation)
+    - Release notes mention the deprecation.
+    - There are no functional changes.
+    - By default, there will be no warnings emitted for such deprecations at runtime.
+
+2. Runtime Deprecation
+
+    - Emits a warning at runtime on first use of the deprecated API or when CLI is used
+    - Can be disabled (environment variable `ZOWE_DEPRECATION_WARN=0` for CLI and  configuration option `zowe.setup.deprecation.warning` in `zowe.yaml` for z/OS Components).
+    - Can be changed to cause an runtime error (by `ZOWE_DEPRECATION_RUNTIME_ERROR=1` and configuration option `zowe.setup.deprecation.runtime-error` in `zowe.yaml` for z/OS Components).
+
+3. End-of-Life
+
+    - Backward-incompatible changes including complete removal of such APIs or functionality can occur at any time.
+
+Apply the `notable change` label to all pull requests that introduce Documentation-Only Deprecations. Such deprecations have no impact on code execution. Thus, they are not breaking changes.
+
+Documentation-Only Deprecations can land in a minor release. They can change to a Runtime Deprecation after six months. They can not change to a End-of-Life until the next major release and need to be in End-of-Life at least for another six months.
+
+No API can change to End-of-Life without going through a Runtime Deprecation cycle. There is no rule that deprecated code must progress to End-of-Life. Documentation-Only and Runtime Deprecations can remain in place for an unlimited duration.
+
+Communicate pending deprecations and associated mitigations with the ecosystem as soon as possible. If possible, do it before the pull request adding the deprecation lands on the master branch.
+
+### Example
+
+1. A planned breaking change is discussed at Zowe architecture call and Zowe playback demo event.
+
+2. The breaking change is brought to TSC and approved.
```

Comment:
I don't think that TSC should approve all breaking changes. This is the responsibility of the squads as long as the change doesn't get to the customers before next major release.

Reply:
Agree like with the previous comment about TSC decisions. I will reword it.

Reply:
Change has been done.